analyses/snv-frequencies/01-snv-frequencies.R

```diff
@@ -0,0 +1,1037 @@
+  # check input parameters
+  stopifnot(is.character(ss_cancer_group))
+  stopifnot(identical(length(ss_cancer_group), as.integer(1)))
+  stopifnot(identical(sum(is.na(ss_cancer_group)), as.integer(0)))
+
+  stopifnot(is.character(ss_cohorts))
+  stopifnot(!is.null(length(ss_cohorts)))
+  stopifnot(identical(length(ss_cohorts), length(unique(ss_cohorts))))
+  stopifnot(identical(sum(is.na(ss_cohorts)), as.integer(0)))
+  stopifnot(length(ss_cohorts) >= 1)
+
+  stopifnot(is.character(valid_url_case_set_ids))
+
+  # a = annotated
+  ss_case_set_id <- get_pcb_pot_csi(ss_cancer_group, ss_cohorts)
+  if (ss_case_set_id %in% valid_url_case_set_ids) {
+    a_mut_freq_tbl <- mut_freq_tbl %>%
+      mutate(PedcBio_PedOT_oncoprint_plot_URL = get_pcb_pot_plot_url(
+        Gene_symbol, ss_case_set_id, 'oncoprint')) %>%
+      mutate(PedcBio_PedOT_mutations_plot_URL = get_pcb_pot_plot_url(
+        Gene_symbol, ss_case_set_id, 'mutations'))
+  } else {
+    a_mut_freq_tbl <- mut_freq_tbl %>%
+      mutate(PedcBio_PedOT_oncoprint_plot_URL = '') %>%
+      mutate(PedcBio_PedOT_mutations_plot_URL = '')
+  }
+
+  return(a_mut_freq_tbl)
+}
+
+
+
+# Generate variant-level mutation frequency table for a (cancer_group, cohort)
+#
+# Args:
+# - maf_df: a MAF tibble. Must contain the following fields:
+#   Kids_First_Biospecimen_ID,
+#   Variant_ID,
+#   Hugo_Symbol,
+#   Gene_full_name,
+#   dbSNP_RS,
+#   IMPACT,
+#   SIFT,
+#   PolyPhen,
+#   Variant_Classification,
+#   Variant_Type,
+#   Protein_RefSeq_ID,
+#   Gene,
+#   ENSP,
+#   HGVSp_Short,
+#   HotSpotAllele.
+# - overall_histology_df: the histology tibble that contains all samples. Must
+#   contain the following fields: Kids_First_Biospecimen_ID,
+#   Kids_First_Participant_ID, cancer_group, and cohort. This
+#   is used for computing the following columns: Total_mutations,
+#   Patients_in_dataset, Total_mutations_Over_Patients_in_dataset and
+#   Frequency_in_overall_dataset.
+# - primary_histology_df: the histology tibble that contains primary tumor
+#   samples. Must contain the Kids_First_Biospecimen_ID field.
+# - relapse_histology_df: the histology tibble that contains relapse tumor
+#   samples. Must contain the Kids_First_Biospecimen_ID field.
+# - ss_cancer_group: a character value of the cancer group to compute for.
+# - ss_cohorts: a vector of character values of the cohorts to compute for.
+# - valid_url_case_set_ids: a vector of character values that are valid
+#   case_set_ids for generating PedcBio PedOT oncoprint and mutations plot URLs.
+#
+# Returns a MAF tibble with additional frequency columns.
+get_cg_ch_var_level_mut_freq_tbl <- function(maf_df, overall_histology_df,
+                                             primary_histology_df,
+                                             relapse_histology_df,
+                                             ss_cancer_group, ss_cohorts,
+                                             valid_url_case_set_ids) {
```

Comment:
For now, we can leave this in, but I am wondering if since we have the gene level table, we should remove the gene level plotting in the variant level table because it may be confusing since it is not the exact set of samples represented in the percentage in the table. We can ask tomorrow.

Reply:
I will keep a note of this. If necessary, I will remove the pedcbio plotting URLs from the variant-level tables in the upcoming snv-frequencies v7 update PR.